### PR TITLE
refactor Dialog to use <dialog> internally

### PR DESCRIPTION
**Pull request by keithamus**
_Reviewers: By approving this Pull Request you are approving the code change, as well as its deployment and mitigation plans._
_Please read this description carefully. If you feel there is anything unclear or missing, please ask for updates._

### What are you trying to accomplish?

This PR removes the `modal-dialog` element we had created to "ponyfill" the native dialog element. Now that native `<dialog>` is well supported, we can simply use that and remove the vast majority of our code.

We still need a `<dialog-helper>` element for 2 features:

 - Our `dialog-show/close-id` attributes. These will be resolved one day by Invokers (https://github.com/whatwg/html/pull/9841)
 - Closing the dialog on backdrop click. This will be one day resolved by https://github.com/openui/open-ui/issues/834

### Screenshots

It is identical in appearance.

### Integration

Yes there is a subtle difference which we'll need to change code in dotcom for.

Our `modal-dialog` element mistakenly dispatches the `cancel` _or_ `close` events on a dialog. This

#### Risk Assessment
  

- [x] **Low risk** the change is small, highly observable, and easily rolled back.
- [ ] **Medium risk** changes that are isolated, reduced in scope or could impact few users. The change will not impact library availability.
- [ ] **High risk** changes are those that could impact customers and SLOs, low or no test coverage, low observability, or slow to rollback.

### Accessibility

- **Fixes axe scan violation** - This change fixes an existing [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation.
- **No new axe scan violation** - This change does not introduce any new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violations.
- **New axe violation** - This change introduces a new [axe scan](https://thehub.github.com/epd/engineering/dev-practicals/frontend/accessibility/readiness-routine/development/#axe-scans) violation. Please describe why the violation cannot be resolved below.

### Merge checklist

- [x] Added/updated tests
- [ ] Added/updated documentation
- [x] Added/updated previews (Lookbook)
- [x] Tested in Chrome
- [x] Tested in Firefox
- [x] Tested in Safari
- [ ] Tested in Edge

Take a look at the [What we look for in reviews](https://github.com/primer/react/blob/main/contributor-docs/CONTRIBUTING.md#what-we-look-for-in-reviews) section of the contributing guidelines for more information on how we review PRs.

### CODE REVIEW

File: app/components/primer/alpha/dialog.rb [hunk 1]
```diff
@@ -125,8 +125,7 @@ def initialize(
         @position_narrow = position_narrow
         @visually_hide_title = visually_hide_title
 
-        @system_arguments[:tag] = "modal-dialog"
-        @system_arguments[:role] = "dialog"
+        @system_arguments[:tag] = "dialog"
```
Comment (camertron) [inline app/components/primer/alpha/dialog.rb:128]: Why is a role no longer necessary?

**Review by camertron — COMMENTED** [review]
This looks good to me, although I will admit I don't know much about the native `<dialog>` element. Since it's newer, should we be concerned we'll run into the same number of issues we ran into with popover?

File: app/components/primer/alpha/dialog.rb (hunk 1 above)
Comment (keithamus) [reply app/components/primer/alpha/dialog.rb:128]: `<dialog>` has the implicit aria role of `dialog`. `modal-dialog` (being a custom element) had no implicit role.

**Comment by keithamus** [comment]
> Since it's newer, should we be concerned we'll run into the same number of issues we ran into with popover?

`<dialog>` is [fully shipped in all browsers](https://caniuse.com/dialog), and we won't require a polyfill. So I don't think so.

**Review by camertron — APPROVED** [review]

**Comment by camertron** [comment]
One last thing: the screenshots are showing a different text color for the body, is that intentional? Approving for now 👍

**Review by langermank — APPROVED** [review]
Looks like the visual changes are gone 😄

**Comment by camertron** [comment]
Sadly I had to revert this in https://github.com/primer/view_components/pull/2468

**Pull request merged**